### DISCUSSION TITLE
Removing self update plugin for packaging safe and future proof?

### DISCUSSION THREAD

**Discussion by thilobillerbeck** [pocketbase/pocketbase#2588] [Q&A]
Hi,

we are currently in the process of upgrading pocketbase to 0.16.x for nixpkgs. While packaging, we thought that removing the internal update functionality would reduce confusion since this is done by nix itself, which also links to non standard library paths. So a self update with a version installed through nix would fail anyway.

During the review (https://github.com/NixOS/nixpkgs/pull/233615) the question came up if this would break in the future or if this plugin was intended to be left inside the main go file. What's your point on that?

**Comment by ganigeorgiev** [comment]
It is safe to remove the self update plugin. it is not required for the binary to work.

Whether the patch approach is future proof - this I guess would depends (_keep in mind that I'm not familiar with Nix and NixOS_). I'm currently planning an internal refactoring to make preparation for the JS app hooks integration and eventually fix some inconsistencies throughout the codebase, so the `examples/base/main.go` may change a little, meaning that you may have to update the patch in the future to reflect the change.

Another more future-proof approach could be to show an error/info message when the user attempts to call the `update` command when the binary is running in Nix env (_and maybe add similar check when inside Docker?_). The only issue with this is that I'm not sure how exactly to identify whether the package was installed with Nix and how reliable that would be. For linux I guess we could use something like `cat /etc/os-release` or similar to get the OS name, but I have to experiment with it. I've added it my todo and will consider it for the next v0.17/v0.18 release.

**Comment by ganigeorgiev** [comment]
@thilobillerbeck _Just to let you know that I've submitted a [minor change in the `develop` branch](https://github.com/pocketbase/pocketbase/commit/e6a41773ca2c44a5fb9cb7095bc2ec55e32f8f83#diff-b712218032499e0927e62b469df4f1e526317e68cd94b413a3cf2a9b8ec30462R405-R408) with a very naive check in case it is running in NixOS env but I haven't tested it as I haven't got the time to really play with Nix yet. If you know a better and more reliable way to perform the check, feel free to let know and I'll update it._
